05_syntax_lab/04.py

```diff
@@ -1,8 +1,22 @@
-"""
-Write a program that reads lines from the user
-until an empty line is inserted.
-After the user typed in an empty line,
-print all previously inserted lines in reverse
-order (from last to first)
-"""
-
+"""
+Elad Agam
+04 - Read lines untill getting an empty lines
+Print all the input line by line - backwards
+"""
+
+Input = []
+while True:
+	print "Enter a line"
+	line = raw_input()
+	if not line == "":
+		Input.append(line)
+	else:
+		break
+
+
+numoflines =(len(Input)) 
+for i in range(numoflines):
+	print Input.pop()
+
+
+

```

Comment:
Done

On 4 July 2016 at 15:38, Gabor Szabo <notifications@github.com> wrote:

> In 05_syntax_lab/04.py
> <https://github.com/ynonp/python-examples-verint-2016-07/pull/5#discussion_r69451784>
> :
>
> > +Input = []
> > +while True:
> > +	print "Enter a line"
> > +	line = raw_input()
> > +	if not line == "":
> > +		Input.append(line)
> > +	else:
> > +		break
> > +
> > +
> > +numoflines =(len(Input))
> > +for i in range(numoflines):
> > +	print Input.pop()
> > +
> > +
> > +
>
> Nice! Especially the use of lists.
> It would be better to use all-lower-case variable names but not input as
> that is a reserved word. Maybe line ?
> It might be better to print the lines without destroying the Input. Can
> you do that?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ynonp/python-examples-verint-2016-07/pull/5/files/ada651d4cf04a795b004e1d13026b76afccb1172#r69451784>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARMSUBTwrlD0DrKYHJUYAeKHJiBGh4Zpks5qSP7XgaJpZM4JEPzj>
> .
>